### DISCUSSION TITLE
Live migration CNI to cilium

### DISCUSSION THREAD

**Discussion by itmwiw** [okd-project/okd#1565] [Q&A]
Hello,
I would like to know if it is possible to migrate existing OKD clusters to cilium cni. In the documentation, I only found how to migrate between OpenshiftOVN and OpenshiftSDN, but nothing about third party CNIs.
Can you help me with this?
Thanks

**Comment by SISheogorath** [comment]
Looking into the details of how the migrations work, it's currently not supported by CNO to migrate to other SDNs than the ones mentioned.

However, jetstack has layed down how they migrated to cilium in a detailed blog post using multus. And multus is already part of OpenShift.

https://www.jetstack.io/blog/cni-migration/

They even put the whole thing in code: https://github.com/jetstack/cni-migration

While I wouldn't recommend to throw this CLI at OpenShift, I would recommend to dive into the details and try to.migrate based on the knowledge (and maybe contribute OpenShift support).

**Comment by vrutkovs** [comment]
Check out this guide - https://veducate.co.uk/migrate-red-hat-openshiftsdn-ovn-kubernetes-cilium/
